sway-lib-std/src/vec.sw

```diff
@@ -2,8 +2,11 @@ library vec;
 
 use ::alloc::{alloc, realloc};
```

Comment:
This has no tests to back up its functionality.
In sway-apps I've mentioned writing tests at the end once you are sure that the code is good however at the same time making a PR requires tests to prove that the code is good so you also need to consider writing tests before submitting new functionality and not just relying on the review, which may be a shallow overview, of the reviewers (because we may not discover a bug which your test might).

Furthermore, no documentation in the code or sway book

Reply:
#2047